EELS/EELS_elemental_mapping.ipynb

```diff
@@ -0,0 +1,789 @@
+    "\n",
+    "m.set_signal_range(1000., 1300.)\n",
+    "m.components.Zn_L1.active = True\n",
+    "m.components.Zn_L2.active = True\n",
+    "m.components.Zn_L3.active = True\n",
+    "\n",
+    "m.multifit(bounded=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Lastly, the power law, the Cu and Zn components are fitted together over the whole energy range."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m.components.PowerLaw.set_parameters_free(parameter_name_list=['A', 'r'])\n",
+    "m.components.Cu_L3.intensity.free = True\n",
+    "m.reset_signal_range()\n",
+    "m.multifit(bounded=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's see the results. Go to the region around `(32, 14)`, where there clearly is copper or zinc of some kind."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m.plot(plot_components=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 3. <a id='save_restore'></a>Saving and restoring the models"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As the fitting process can be slow, saving the models can be a good idea. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m.save('model.hspy', overwrite=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This results in a file called `model.hspy`, which can be loaded and restored."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mr = hs.load('model.hspy')\n",
+    "mr = mr.models.restore('a')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mr.plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true,
+    "jupyter": {
+     "outputs_hidden": true
+    }
+   },
+   "source": [
+    "# 4. <a id='plotting'></a>Make maps"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this section, elemental maps will be plotted from the intensity of the model of the Cu-L$_3$ and Zn-L$_3$ components saved above. The HAADF image will also be plotted, for comparison with the elemental distribution.\n",
+    "\n",
+    "**Note:** as the components L$_3$, L$_2$ and L$_1$ are connected during the fitting procedure, we only need to visualize one of them to get a relative elemental map."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
```

Comment:
I added a section on how to show the results using `m.components.Zn_L3.intensity.plot()` and `hs.plot.plot_images`.

I think having the more advanced plotting, via matplotlib, is also useful. As people often want to create high quality figures from the data they've processed, and having it directly in the same notebook makes it easier to do this.

But it can of course be moved to a separate notebook, ideally in the same `exspy-demos/EELS/` folder.

Reply:
I think this is important to split them in separate notebook:
- the elemental mapping notebook will be more clear and more focused (less clutter due to the plotting code). One advantage of using hyperspy/exspy is remove a lot of the python code in workflows and the tutorial should reflect this.
- The notebook on customising matplotlib figure will get more visibility on its own rather than included in EELS elemental mapping notebook. Also since this can be used for EDS maps in a very similar fashion, it should go in somewhere else than the EELS folder so that user interested in EDS can also find this notebook easily.
- At the end of the day, users will copy and paste code to make their own notebook, so it doesn't need to be in the same notebook?

Reply:
Good points! I moved the matplotlib plotting part to its own notebook.

I created a new folder `plotting_and_visualization`, but not sure if that is the best name.

In addition, should the `model.hspy` file be added as well? For example in `EELS/datasets/`? This so people don't have to run through the whole EELS elemental map notebook before they can run the plotting notebook. It is 1.3 MB.